### PR TITLE
feat: Create package for mobile components

### PR DESCRIPTION
**Pull request by rodrigoeidelvein**
## Motivations
To unify our design system, we'll now have a package for the mobile components! For this first version, I imported `Icon` and `Divider`. 

### Added

- `components-mobile` package!
- `Icon` and `Divider` components

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

You can pack this package and install it on jobber mobile.
Assuming you have `atlantis` and `jobber-mobile` in the same folder:
On atlantis root:
`npm pack -w @jobber/components-native`

After packing it, you can run on jobber-mobile root:
`npm install ../atlantis/jobber-components-native-0.0.0.tgz`

Also, I created a draft PR with the prerelease installed in jobber-mobile: https://github.com/GetJobber/jobber-mobile/pull/7212 and there you can download the builds from Bitrise

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: docs/guides/index.tsx
Comment (darryltec) [inline docs/guides/index.tsx:1]: I know i added this, but we don't need this. It's for testing purposes 😆

File: .eslintrc.js [hunk 1]
```diff
@@ -3,12 +3,14 @@ require("@jobber/eslint-config/patch-eslint-plugin-resolution.js");
 
 const packageAliases = [
   ["@jobber/components", "./packages/components/src"],
+  ["@jobber/components-native", "./packages/components-native/src"],
```
Comment (darryltec) [inline .eslintrc.js:6]: are we going for 

```js
import { Icon } from "@jobber/components-native"
```

if so, we don't need this.

(PS: I also added this, I'm sorry 😬 )

File: package.json [hunk 2]
```diff
@@ -60,6 +61,7 @@
     "@babel/preset-typescript": "^7.3.3",
     "@csstools/postcss-global-data": "^1.0.3",
     "@jobber/components": "*",
+    "@jobber/components-native": "*",
```
Comment (darryltec) [inline package.json:64]: we probably shouldn't install it yet since we don't use em right now.

File: packages/components-native/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+## Usage
+
+Import and render an Atlantis component:
+
+```tsx
+import React from "react";
+import { Icon } from "@jobber/components-native";
+
+export default function MyComponent() {
```
Comment (darryltec) [inline packages/components-native/README.md:28]:
```suggestion
export function MyComponent() {
```

because we don't really do default in the jobber

File: packages/components-native/src/Divider/Divider.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { View } from "react-native";
+import { styles } from "./Divider.style";
+
+interface DividerProps {
+  /**
+   * The weight of the divider.
+   *
+   * @default "base"
+   */
+  readonly size?: "base" | "large" | "largest";
```
Comment (darryltec) [inline packages/components-native/src/Divider/Divider.tsx:11]: I wonder if

```suggestion
  readonly size?: keyof typeof styles;
```

would work here 🤔

File: tsconfig.json
Comment (darryltec) [inline tsconfig.json:1]: I think you can revert the changes on this file again. it was added to support `@jobber/components-native/Icon` which we don't do anymore, yeah?

**Review by darryltec — COMMENTED** [review]
Few little bits for you mah dude. Mostly reversing the silly things I added haha

**Comment by rodrigoeidelvein** [comment]
I got into this weird state when using `InputPhoneNumber` in a `InputGroup`:
image

File: package.json (hunk 2 above)
Comment (MichaelParadis) [reply package.json:64]: Do we need `esm` for `components-native`? Wouldn't this affect the storybook stuff if it isn't installed?

Comment (darryltec) [reply package.json:64]: it won't since we don't reference it in the docs. It's esm but still a js file not mjs.

Comment (darryltec) [reply package.json:64]: as for

> Do we need esm 

If you look at react native's package, they use esm instead of common js. Not sure of RN tree shake things but esm is what's tree shakeable

File: packages/components-native/src/Divider/Divider.tsx (hunk 4 above)
Comment (MichaelParadis) [reply packages/components-native/src/Divider/Divider.tsx:11]: That would key rid of `base` and add `divider`. We may be able to do something like
```
export const styles = StyleSheet.create({
  divider: {
    height: JobberStyle["space-minuscule"],
    margin: 0,
    borderBottomWidth: JobberStyle["border-base"],
    borderBottomColor: JobberStyle["color-border"],
  },
  sizes: {
    large: {
      borderBottomWidth: JobberStyle["border-thick"],
      opacity: 0.875,
    },
    largest: {
      borderBottomWidth: JobberStyle["space-small"],
      opacity: 0.375,
    },
  }
});
```

```
readonly size?: keyof typeof styles.sizes;
```
Though

File: package.json (hunk 2 above)
Comment (MichaelParadis) [reply package.json:64]: Ahh, good point the storybook stuff is in a separate issue

Also makes sense about the RN package stuff. Thanks for the clarification

File: package.json [hunk 5]
```diff
@@ -113,6 +115,7 @@
     "prop-types": "^15.7.2",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-native": "^0.71.7",
```
Comment (MichaelParadis) [inline package.json:118]: I think we should match the Jobber mobile version instead of this version

File: packages/components-native/package.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    "@types/node": "*",
+    "@types/react": "*",
+    "@types/react-native": "*",
+    "metro-react-native-babel-preset": "^0.76.0",
+    "typescript": "*"
+  },
+  "peerDependencies": {
+    "@babel/core": "*",
+    "react": "*",
+    "react-native": "*"
```
Comment (MichaelParadis) [inline packages/components-native/package.json:38]: I am hesitant about an unpinned dependency here. If when packaging it grabs from the main package.json this is fine but if not we should probably have a version specified

File: packages/components-native/package.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  "dependencies": {
+    "@jobber/design": "*",
+    "react-native-svg": "^12.4.4",
+    "ts-xor": "^1.1.0"
+  },
+  "devDependencies": {
+    "@types/node": "*",
+    "@types/react": "*",
+    "@types/react-native": "*",
```
Comment (MichaelParadis) [inline packages/components-native/package.json:31]: Same hesitation as below types that don't match the RN version being used can cause issues. If it would grab it from the parent package.json I would specify a version there

The same applies for React versions here too

File: packages/components-native/package.json (hunk 6 above)
Comment (MichaelParadis) [reply packages/components-native/package.json:38]: Also applies to the react version

File: packages/components-native/package.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    "metro-react-native-babel-preset": "^0.76.0",
+    "typescript": "*"
+  },
+  "peerDependencies": {
+    "@babel/core": "*",
+    "react": "*",
+    "react-native": "*"
+  },
+  "eslintIgnore": [
```
Comment (MichaelParadis) [inline packages/components-native/package.json:40]: Isn't this not needed because of the `.eslintignore` file in the main repo?

File: packages/components-native/tsconfig.types.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./dist/types",
```
Comment (MichaelParadis) [inline packages/components-native/tsconfig.types.json:4]: I think this file can be moved into the one `tsconfig` file using this tsconfig option
https://www.typescriptlang.org/tsconfig#declarationDir

File: packages/components-native/src/Divider/Divider.tsx (hunk 4 above)
Comment (MichaelParadis) [reply packages/components-native/src/Divider/Divider.tsx:11]: This would work though

```diff
diff --git a/packages/components-native/src/Divider/Divider.style.ts b/packages/components-native/src/Divider/Divider.style.ts
index 72c1d6ce..001d218e 100644
--- a/packages/components-native/src/Divider/Divider.style.ts
+++ b/packages/components-native/src/Divider/Divider.style.ts
@@ -9,6 +9,9 @@ export const styles = StyleSheet.create({
     borderBottomWidth: JobberStyle["border-base"],
     borderBottomColor: JobberStyle["color-border"],
   },
+});
+
+export const sizeStyles = StyleSheet.create({
   large: {
     borderBottomWidth: JobberStyle["border-thick"],
     opacity: 0.875,
diff --git a/packages/components-native/src/Divider/Divider.tsx b/packages/components-native/src/Divider/Divider.tsx
index 03ebf72f..efe09d06 100644
--- a/packages/components-native/src/Divider/Divider.tsx
+++ b/packages/components-native/src/Divider/Divider.tsx
@@ -1,6 +1,6 @@
 import React from "react";
 import { View } from "react-native";
-import { styles } from "./Divider.style";
+import { sizeStyles, styles } from "./Divider.style";
 
 interface DividerProps {
   /**
@@ -8,14 +8,14 @@ interface DividerProps {
    *
    * @default "base"
    */
-  readonly size?: "base" | "large" | "largest";
+  readonly size?: "base" | keyof typeof sizeStyles;
 }
 
 export function Divider({ size = "base" }: DividerProps): JSX.Element {
   const style = [
     styles.divider,
-    size === "large" && styles.large,
-    size === "largest" && styles.largest,
+    size === "large" && sizeStyles.large,
+    size === "largest" && sizeStyles.largest,
   ];
   return <View testID="Divider" style={style} />;
 }

```

Comment (rodrigoeidelvein) [reply packages/components-native/src/Divider/Divider.tsx]: What if we just change the `divider` property to `base`? See the new changes

File: package.json (hunk 5 above)
Comment (rodrigoeidelvein) [reply package.json:117]: Good idea!

File: packages/components-native/tsconfig.types.json (hunk 9 above)
Comment (rodrigoeidelvein) [reply packages/components-native/tsconfig.types.json]: I also removed `tsconfig.esm.json`.

File: packages/components-native/package.json (hunk 6 above)
Comment (rodrigoeidelvein) [reply packages/components-native/package.json:38]: Do you think we should put the same version as we have in jobber-mobile?

Comment (MichaelParadis) [reply packages/components-native/package.json:38]: I think so. We may want a couple patch versions below but I think matching jobber-mobile for now is good. We don't really want to have things be broken on other versions

Comment (rodrigoeidelvein) [reply packages/components-native/package.json]: I got an peer dependency issue because `react-native@0.69.10` (version installed on jobber-mobile), has a peer dependency on `react@18.0.0` and we currently have `react@18.2` installed.

I think it's safe for us to use the latest version in `components-native`. I added a minimum peer dependency version to be the same one as jobber-mobile

File: packages/components-native/tsconfig.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
+{
+  "extends": "../../tsconfig.base.json",
+  "compilerOptions": {
+    "outDir": "./dist",
```
Comment (MichaelParadis) [inline packages/components-native/tsconfig.json:4]:
```suggestion
    "outDir": "./dist/esm",
```
Or update the package.json

Comment (rodrigoeidelvein) [reply packages/components-native/tsconfig.json:4]: Good catch! I updated the `package.json`

File: packages/components-native/LICENSE [hunk 11]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2019 Octopusapp Inc.
```
Comment (MichaelParadis) [inline packages/components-native/LICENSE:3]: Year should be updated, however they all should be so this one can be done and we can update the other ones in another PR

**Review by MichaelParadis — APPROVED** [review]
LGTM!
